x/posts/internal/keeper/keeper.go

```diff
@@ -139,7 +140,20 @@ func (k Keeper) GetPostsFiltered(ctx sdk.Context, params types.QueryPostsParams)
 			matchCreator = params.Creator.Equals(post.Creator)
 		}
 
-		if matchParentID && matchCreationTime && matchAllowsComments && matchSubspace && matchCreator {
+		// match hashtags if provided
+		if len(params.Hashtags) > 0 {
+			postHashtags := post.GetPostHashtags()
+			matchHashtags = len(postHashtags) == len(params.Hashtags)
+			sort.Strings(postHashtags)
+			sort.Strings(params.Hashtags)
+			if matchHashtags {
+				for index, hashtag := range params.Hashtags {
+					matchHashtags = postHashtags[index] == strings.Trim(hashtag, "#")
```

Comment:
Still need to keep track of iterations. Also, do we really need the trim? I think we can remove it

```suggestion
					matchHashtags &= postHashtags[index] == hashtag
```